alarm/nialarm.yml

```diff
@@ -0,0 +1,686 @@
+
+responses:
+  Error:
+    description: Error
+    schema:
+      description: Error response
+      title: ErrorResponse
+      type: object
+      properties:
+        error:
+          $ref: '#/definitions/Error'
+  QueryResponse:
+    description: Query response
+    schema:
+      description: Query response
+      title: QueryResponse
+      type: object
+      properties:
+        totalCount:
+          description: The total number of alarm instances which matched your query
+          type: integer
+          example: 1
+        filterMatches:
+          description: Array containing the alarm instances which matched your query
+          type: array
+          items:
+            $ref: '#/definitions/AlarmInstance'
+  Unauthorized:
+    description: Not authorized
+    headers:
+      WWW_Authenticate:
+        description: Information for how to authenticate
+        type: string
+paths:
+  /:
+    get:
+      tags: [versioning]
+      summary: API information
+      description: Returns information about API versions and available operations.
+      operationId: RootEndpoint
+      x-ni-request-all-privileges: true
+      # Explicitly mark security as an empty array - this route does not require any privileges.
+      # Marking it this way prevents it from inheriting the top-level security settings.
+      security: []
+      responses:
+        200:
+          description: OK
+          schema:
+            description: Version information
+            title: RootEndpointResponse
+            type: object
+            properties:
+              v1:
+                $ref: '#/definitions/V1Operations'
+              version:
+                description: Implementation version of the web service
+                type: string
+  /{version}:
+    parameters:
+      - in: path
+        name: version
+        description: Version of the API to retrieve operations.
+        type: string
+        required: true
+    get:
+      tags: [versioning]
+      summary: API version information
+      description: Returns available operations for a single version of the API.
+      operationId: RootEndpointWithVersion
+      x-ni-request-all-privileges: true
+      # Explicitly mark security as an empty array - this route does not require any privileges.
+      # Marking it this way prevents it from inheriting the top-level security settings.
+      security: []
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/V1Operations'
+        404:
+          description: Not Found
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/acknowledge-instances:
+    post:
+      tags: [alarmInstances]
+      summary: Acknowledge alarm instances
+      description: Acknowledges one or more alarm instances, optionally forcing them clear and adding notes to them.
+      operationId: AcknowledgeInstances
+      parameters:
+        - in: body
+          name: Request body
```

Comment:
Should this parameter be marked as required?

Reply:
Yep!